lib/internal/Magento/Framework/Mview/View/Changelog.php

```diff
@@ -154,9 +152,39 @@ public function getList($fromVersionId, $toVersionId)
             (int)$toVersionId
         );
 
+        return $select;
+    }
+
+    /**
+     * Retrieve entity ids by range [$fromVersionId..$toVersionId]
+     *
+     * @param int $fromVersionId
+     * @param int $toVersionId
+     * @return int[]
+     * @throws ChangelogTableNotExistsException
+     */
+    public function getList($fromVersionId, $toVersionId)
+    {
+        $select = $this->getListSelect($fromVersionId, $toVersionId);
         return $this->connection->fetchCol($select);
     }
 
+    /**
+     * Retrieve the count of entity ids in the range [$fromVersionId..$toVersionId]
+     *
+     * @param int $fromVersionId
+     * @param int $toVersionId
+     * @return int[]
+     * @throws ChangelogTableNotExistsException
+     */
+    public function getListSize($fromVersionId, $toVersionId)
```

Comment:
This method was just added, this change is not backward compatible. Unfortunately we can't accept  such change. Couldn't it be done without adding new methods?
If no - then we need to introduce new interface that will include be method.

Reply:
Thanks I was waiting for the suite to go green before raising questions about this function, this was my attempt to address this comment: https://github.com/magento/magento2/pull/12122#discussion_r149927089

I don't believe this can be achieved without a similar query, I was trying to prevent the `SELECT` query from being generated in two places, but if the interface is not up for modification I can either do it in a new interface or directly within the `Command` class itself. How do you think this should be architected? If you'd suggest a new interface how do you think it should be named etc?

Reply:
Is this 4af04b1c43ea8b1198ba5027500552c88a09737c what you meant?